pkg/controllers/workapplier/controller.go

```diff
@@ -417,7 +417,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		Kind:               fleetv1beta1.AppliedWorkKind,
 		Name:               appliedWork.GetName(),
 		UID:                appliedWork.GetUID(),
-		BlockOwnerDeletion: ptr.To(false),
+		BlockOwnerDeletion: ptr.To(true),
```

Comment:
Updating this directly may cause old objs not being able to be deleted due to: https://github.com/kubefleet-dev/kubefleet/blob/main/pkg/controllers/workapplier/preprocess.go#L536C1-L552C2 The expectedOwnerReference is not "deepEqual" anymore.

Reply:
Yeah, for the part we might need to exempt the `blockOwnerDeletion` field from being compared.

Reply:
Though I am wondering if K8s allows adding an owner ref with the same UIDs/namespaced keys... will the system auto-merge the entries?